### ISSUE TITLE
NoWhitespaceAfter and NoWhitespaceBefore false positives in method signatures

### ISSUE THREAD

**Issue by beamerblvd** [checkstyle/checkstyle#6390]
I have the following configs:
```xml
<module name="NoWhitespaceAfter">
    <property name="allowLineBreaks" value="false" />
    <property name="tokens" value="AT,BNOT,DEC,INC,LITERAL_SYNCHRONIZED,LNOT" />
</module>
<module name="NoWhitespaceAfter">
    <property name="allowLineBreaks" value="true" />
    <property name="tokens"
              value="ARRAY_DECLARATOR,ARRAY_INIT,DOT,INDEX_OP,METHOD_REF,UNARY_MINUS,UNARY_PLUS" />
</module>
<module name="NoWhitespaceBefore">
    <property name="allowLineBreaks" value="false" />
    <property name="tokens" value="COMMA,DOT,ELLIPSIS,METHOD_REF,POST_INC,POST_DEC,SEMI" />
</module>
<module name="NoWhitespaceBefore">
    <property name="allowLineBreaks" value="true" />
    <property name="tokens" value="GENERIC_END,GENERIC_START" />
</module>
```

These two methods should not trigger either:

```java
       # ImmutableAbstractCollection.java
178    @Override
179    @SuppressWarnings("SuspiciousToArrayCall")
180    public final <T> T[] toArray(final T[] prototype)
181    {
182        synchronized(this.internalCollection)
183        {
184            this.checkValidity();
185            return this.internalCollection.toArray(prototype);
186        }
187    }

       # LicenseManager.java
144    public static synchronized LicenseManager getInstance()
145    {
146        if(LicenseManager.instance == null)
147        {
148            LicenseManager.instance = new LicenseManager();
149        }
150
151        return LicenseManager.instance;
152    }
```

However, I get these errors and have to suppress them:

```
... ImmutableAbstractCollection.java [180,18] (whitespace) NoWhitespaceBefore: '<' is preceded with whitespace.
... LicenseManager.java:[144,19] (whitespace) NoWhitespaceAfter: 'synchronized' is followed by whitespace.
```

If the former callout were changed, it would technically compile, but it would not be correct. If the latter callout were were changed, it would not compile at all. These inspections are not correct.

**Comment by rnveach** [comment]
@beamerblvd Please see http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug.3F .
It is hard to understand what your line numbers are referring to when they don't match up your example.

**Comment by romani** [comment]
Please strictly follow required format. It will give us all required details.
Issues without proper format will be closed in 1 month.

----

Please point me to checkstyle config in your project, if you are using a lot of Checks, we could add your project to our CI

**Comment by romani** [comment]
No reply, mean no interest to cooperate to resolve issue. Closing
